### ISSUE TITLE
Uploading with v2 API

### ISSUE THREAD

**Issue by lcolladotor** [ropensci/rfigshare#96]
Hi,

I've talked with Carl and I know you are quite busy. Which is why I tried myself to figure out a way to upload files using the v2 API by looking at `rfigshare::fs_upload()` and `rfigshare:::fs_upload_one()`. I can get close, but haven't been able to actually upload a file. This is my first time using `httr` directly, so maybe there is something obvious that I'm missing. I'm doing all of this simply because with the v1 API the file size is still limited to 250 mb (I get 413 status codes using `rfigshare::fs_upload` for files larger than 250 mb). I can see that that was the previous limit as shown at http://blog.impactstory.org/impact-challenge-data-repository/ but in theory now the file size limit is 5 GB. My guess is that the file size limit is 5 GB only using the v2 api.

Anyhow, you can see my tests at https://gist.github.com/lcolladotor/dbcb842a931095c25bfa1036c7d08e0b. I was able to authenticate using a "personal token". I can use `httr::POST()` to create the file then `httr::GET()` to find the upload url, which contains the upload token as part of it. However, when I try to `httr::PUT()` I keep getting the 404 status code. That code according to https://docs.figshare.com/api/file_uploader/ means that the upload token is unknown. I've tried adding it to the headers, but that didn't work either. For example:

```
## Attempt 7
out2 <- PUT(content(resp)$upload_url, c(headers, add_headers('upload_token' = content(resp)$upload_token)), body = upload_file(file), encode = 'multipart')
status_code(out2)

## Attempt 8
out2 <- PUT(content(resp)$upload_url, c(headers, add_headers('upload_token' = content(resp)$upload_token)), body = list('filedata' = upload_file(file)), encode = 'multipart')
status_code(out2)
```

I've been trying to follow the code at https://docs.figshare.com/api/upload_example/ and I'm basically stuck at:

```
url = '{upload_url}'.format(**json.loads(resp.content))
parts = json.loads(requests.get(url).content)['parts']

# Upload parts
with open(file_path, 'rb') as fin:
    for part in parts:
        size = part['endOffset'] - part['startOffset'] + 1
        address = '{}/{}'.format(url, part['partNo'])
        requests.put(address, data=fin.read(size))
```

My understanding from looking at `httr` is that I shouldn't have to write a parts uploader. But I do see parts information when I use `httr::GET()` on the upload url. 

Anyhow, I hope that it's something simple to figure out. I'll appreciate any pointers you have on what to try.

Thanks,
Leonardo

**Comment by lcolladotor** [comment]
We got this to work using https://github.com/nellore/runs/blob/master/recount2/figshare.py which is adapted from https://docs.figshare.com/api/upload_example/. Then in R I simply used `system()` calls.

With this workaround I was able to upload files over 250 mb. I even tested one file by re-downloading and checking that the md5sum string was the same one. However, the upload speeds were pretty slow. For a particular 3.7 GB file I gave up after 1 hr. I then downsized my test to a 270 mb file.
